src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -0,0 +1,421 @@
+        {
+            if (!_geometryEditor.IsStarted)
+            {
+                ToolComboBox.IsEnabled = true;
+
+                // Disable buttons to reflect that the geometry editor has started.
+                DisableOtherGeometryButtons(PolylineButton);
+
+                _geometryEditor.Start(GeometryType.Polyline);
+            }
+        }
+
+        // Start the geometry editor with the polygon geometry type.
+        private void PolygonButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (!_geometryEditor.IsStarted)
+            {
+                ToolComboBox.IsEnabled = true;
+
+                // Disable buttons to reflect that the geometry editor has started.
+                DisableOtherGeometryButtons(PolygonButton);
+
+                _geometryEditor.Start(GeometryType.Polygon);
+            }
+        }
+
+        // Set the geometry editor tool from the combo box.
+        private void ToolComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var keyValuePair = (KeyValuePair<string, GeometryEditorTool>)ToolComboBox.SelectedItem;
+            _geometryEditor.Tool = keyValuePair.Value;
+        }
+
+        #endregion Create controls event handlers
+
+        #region Edit controls event handlers
+
+        // Undo the last change made to the geometry while editing is active.
+        private void UndoButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Undo();
+        }
+
+        // Redo the last change made to the geometry while editing is active.
+        private void RedoButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.Redo();
+        }
+
+        // Delete the currently selected element of the geometry editor.
+        private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.DeleteSelectedElement();
+        }
+
+        // Update an existing graphic or create a new graphic.
+        private void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_selectedGraphic != null)
+            {
+                // Update the geometry of the graphic being edited and make it visible again.
+                _selectedGraphic.Geometry = _geometryEditor.Stop();
+                _selectedGraphic.IsVisible = true;
+
+                // Reset the selected graphic.
+                _selectedGraphic.IsSelected = false;
+                _selectedGraphic = null;
+            }
+            else
+            {
+                _geometryCreated = true;
+
+                // Get the geometry from the geometry editor and create a new graphic for it.
+                Geometry geometry = _geometryEditor.Stop();
+                var graphic = new Graphic(geometry);
+
+                // Set the graphic style based on geometry type.
+                switch (geometry.GeometryType)
```

Comment:
I wonder if this can be refactored into a `GetSymbol(..)` method so that it does not look like you have to do a lot to add a new graphic from the editor. I can't remember if there's a chance geometry result is null, maybe a null-check before getting it's type and adding to the overlay?

```CSharp
var geometry = MyMapView.GeometryEditor.Stop();
var symbol = GetSymbol(geometry);
MyMapView.GraphicsOverlay[0].Graphics.Add(new Graphic(geometry, symbol));
```

I think the if/else can even be combined a little.

```CSharp
var geometry = MyMapView.GeometryEditor.Stop();
if(_selectedGraphic is not null)
{
  _selectedGraphic.Geometry = geometry;
  ResetSelectedGraphic();
}
else
{
  var symbol = GetSymbol(geometry);
   MyMapView.GraphicsOverlay[0].Graphics.Add(new Graphic(geometry, symbol));
}
```

You may need to reset when identify fails, save, discard, no?
```CSharp
        private void ResetSelectedGraphic()
        {
            if (_selectedGraphic is not null)
            {
                _selectedGraphic.IsSelected = false;
                _selectedGraphic.IsVisible = true;
            }
            _selectedGraphic = null;
        }
```